src/cgroup.c

```diff
@@ -279,19 +279,35 @@ gboolean check_cgroup2_oom()
 	ssize_t read;
 	while ((read = getline(&line, &len, fp)) != -1) {
 		long int counter;
+		int memcmp_oom, memcmp_oom_kill, len;
+		const int oom_len = 4, oom_kill_len = 9;
 
-		if (read < 6 || memcmp(line, "oom ", 4))
+		if (read < 6)
 			continue;
 
-		counter = strtol(&line[4], NULL, 10);
+		memcmp_oom = memcmp(line, "oom ", oom_len);
+		memcmp_oom_kill = memcmp(line, "oom_kill ", oom_kill_len);
```

Comment:
we risk to read after the line length.

The `read < 6` protects only for `oom `.